### ISSUE TITLE
feat[dev]: ✨ implement A11y Re-Improvements to sd-dialog

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#2073]
## Description
This tasks describes the Figma implementation for: #2048 

This task describes the Storybook implementation for the sd-dialog component as part of the A11y re-improvements outlined in Epic #2048.

A11y improvements for sd-dialog can be found here: [sd-dialog Notes](https://www.notion.so/matuzo/sd-dialog-1baf3f6e1bc0819199f6f28805070d1d).

------------------------------------------

## Comment
/

## Notes
- The UX in timed dialogs would be better if the disabled button was focusable (see [[sd-button](https://www.notion.so/sd-button-1baf3f6e1bc08198a2d6ddf4df8b1e21?pvs=21)](https://www.notion.so/sd-button-1baf3f6e1bc08198a2d6ddf4df8b1e21?pvs=21)). You could use `aria-describedby` on the button and create a reference to the countdown. That way, the user would know why the button is disabled.

```html
<div id="countdown">
  Closable in 4 seconds...
</div>

<button aria-disabled="true" aria-describedby="countdown">
	Close
</button>
```

This example is currently on under the sd-dialog screenshots "Preventing closing".

**Solution:**
- Move the "Preventing closing" screenshot to a template and implement it using correct accessibility patterns. (Visually disabled, aria-describedby..)

- Please re-check the `aria-label` and `aria-labelledby`. Whenever the `headline` attribute is not set, check if aria-label is still rendered and if it being overriding `aria-labelledby`. It shouldn't be the case.
When the `headline` attribute is not set, aria-labelledby should be the one announced.

## Accessibility information
- Always provide a headline for the dialog.
- *Move the keyboard focus to the first interactive element within the modal (usually the close button) when the modal opens. Keep the keyboard focus within the modal while it is open, and return focus to the element that triggered the modal when it is closed.* **(To be removed on figma & overview pages)**

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
